ui/src/app/machines/views/MachineList/OwnerColumn/OwnerColumn.js

```diff
@@ -1,24 +1,68 @@
-import { useSelector } from "react-redux";
-import React from "react";
+import { Loader } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+import React, { useEffect, useState } from "react";
 import PropTypes from "prop-types";
 
+import { machine as machineActions } from "app/base/actions";
 import { machine as machineSelectors } from "app/base/selectors";
 import DoubleRow from "app/base/components/DoubleRow";
 
 const OwnerColumn = ({ onToggleMenu, systemId }) => {
+  const dispatch = useDispatch();
+  const [updating, setUpdating] = useState(null);
   const machine = useSelector(state =>
     machineSelectors.getBySystemId(state, systemId)
   );
 
   const owner = machine.owner ? machine.owner : "-";
   const tags = machine.tags ? machine.tags.join(", ") : "";
+  const hasAcquireAction = machine.actions.includes("acquire");
+  const hasReleaseAction = machine.actions.includes("release");
+  let menuLinks = [];
```

Comment:
I wonder if this whole section of creating the menu links could be generalised somehow? Like a custom hook that you just pass in the allowed actions, something like `const menuLinks = useTableMenuLinks(["acquire", "release"])`. :man_shrugging:

Reply:
I've added an issue #793